src/views/states/list/hooks/useSortStates.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+import { ThSortType } from '@patternfly/react-table/dist/esm/components/Table/base';
+import { V1beta1NodeNetworkState } from '@types';
+
+const useSortStates = (
+  nodeNetworkStates: V1beta1NodeNetworkState[],
+): { sortedStates: V1beta1NodeNetworkState[]; nameSortParams: ThSortType } => {
+  const [activeSortIndex, setActiveSortIndex] = useState<number | null>(null);
+  const [activeSortDirection, setActiveSortDirection] = useState<'asc' | 'desc' | null>(null);
+
+  let sortedStates = nodeNetworkStates;
+  if (activeSortIndex === 1) {
+    sortedStates = nodeNetworkStates.sort((a, b) => {
+      const aValue = a.metadata.name;
+      const bValue = b.metadata.name;
+
+      if (activeSortDirection === 'asc') {
+        return (aValue as string).localeCompare(bValue as string);
+      }
+      return (bValue as string).localeCompare(aValue as string);
+    });
+  }
+
+  const nameSortParams: ThSortType = {
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
```

Comment:
Maybe we should add types for `event`, `index` and `direction`?

Reply:
Im with hilda here, i like also to see the types directly instead of searching them on component docs, or hovering :)